src/Compiler/AbstractIL/ilwrite.fs

```diff
@@ -2866,6 +2869,14 @@ let rec GenTypeDefPass3 enc cenv (tdef: ILTypeDef) =
    try
         let env = envForTypeDef tdef
         let tidx = GetIdxForTypeDef cenv (TdKey(enc, tdef.Name))
+
+        match tdef.ImplementsCustomAttrs with
+        | None -> ()
+        | Some attrList ->
+            attrList
+            |> List.zip cenv.implementsIdxs[tidx]
+            |> List.iter (fun (impIdx,(attrs,metadataIdx)) -> GenCustomAttrsPass3Or4 cenv (hca_InterfaceImpl,impIdx) (attrs.GetCustomAttrs metadataIdx))
```

Comment:
What is the recommended approach in such cases for the best performance?
`List.zip` creates a list which is immediately garbage collected.
Isn't it more efficient to call `Seq.zip` followed by `Seq.iter`?

Reply:
You are right, this was laziness on my side.

There is `List.iter2`  iterates two lists simulatenously without allocations (even without the implicit one used when `Seq` is involved).